src/System.Text.Utf8String/System/Text/Utf8Span.CodePointReverseEnumerator.cs

```diff
@@ -0,0 +1,121 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+
+namespace System.Text.Utf8
+{
+    partial struct Utf8Span
+    {
+        // TODO: Name TBD
+        public ref struct CodePointReverseEnumerator
+        {
+            private ReadOnlySpan<byte> _buffer;
+            private int _index;
+            private int _currentLenCache;
+            private const int ResetIndex = -Utf8Helper.MaxCodeUnitsPerCodePoint - 1;
+
+            public unsafe CodePointReverseEnumerator(ReadOnlySpan<byte> buffer) : this()
```

Comment:
This method (amongst others) doesn't need to be unsafe.

Reply:
@A-And, is that something you still want to change?